zilencer/views.py

```diff
@@ -158,33 +162,52 @@ def remote_server_notify_push(request: HttpRequest, entity: Union[UserProfile, R
 @zulip_login_required
 def initial_upgrade(request: HttpRequest) -> HttpResponse:
     user = request.user
+    error_message = ""
+
     if Customer.objects.filter(realm=user.realm).exists():
         return HttpResponseRedirect(reverse('zilencer.views.billing_home'))
 
     if request.method == 'POST':
-        stripe_customer = do_create_customer_with_payment_source(user, request.POST['stripeToken'])
-        # TODO: the current way this is done is subject to tampering by the user.
-        seat_count = int(request.POST['seat_count'])
-        if seat_count < 1:
-            raise AssertionError('seat_count is less than 1')
-        do_subscribe_customer_to_plan(
-            stripe_customer=stripe_customer,
-            stripe_plan_id=Plan.objects.get(nickname=request.POST['plan']).stripe_plan_id,
-            seat_count=seat_count,
-            # TODO: billing address details are passed to us in the request;
-            # use that to calculate taxes.
-            tax_percent=0)
-        # TODO: check for errors and raise/send to frontend
-        return HttpResponseRedirect(reverse('zilencer.views.billing_home'))
+        signed_seat_count = request.POST['signed_seat_count']
+        salt = request.POST['salt']
+
+        try:
+            seat_count = int(unsign_string(signed_seat_count, salt))
+            if seat_count < 1:
```

Comment:
We can remove this now that it's signed; it was to project against injection attacks or weird things like someone putting a negating seat count and us paying them money instead of getting money :).

Reply:
Done.